### ISSUE TITLE
[BUG] Modals do not place focus back on activating element

### ISSUE THREAD

**Issue by RasmusKjeldgaard** [kirbydesign/designsystem#3963]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug
Turns out the functionality for focusing a modals title introduced in https://github.com/kirbydesign/designsystem/pull/3927 focussed the title before ionic had successfully stored a reference to the activating element, which should be focussed again once the modal is dismissed.

## Describe how to reproduce the bug

1. Go to https://stackblitz.com/edit/stackblitz-starters-mqmytrr6?file=src%2Fmain.ts
2. Click button 'Show Modal'
3. Close modal again (<kbd>esc</kbd> or close button)
4. Observe that the active element is now body, instead of the button that activated the modal

## Which Kirby version was used?
Currently unreleased code, but can be seen in v10.6.0-rc.0

## What was the expected behavior?
Focus should be restored to activating element.
<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/develop/.github/SUPPORT.md#team-kirby); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.
